org.adempiere.ui.zk/WEB-INF/src/web/js/org/idempiere/commons/input.js

```diff
@@ -45,6 +45,61 @@ zk.afterLoad(function() {
     });
 });
 
+zk.afterLoad('zk', function() {
+	// https://www.zkoss.org/wiki/ZK_Client-side_Reference/General_Control/Widget_Customization#Override_a_Default_Widget_Method_in_JavaScript_File
+	var exWidget = {};
+	// https://www.zkoss.org/javadoc/latest/jsdoc/functions/zk.html#override
+	zk.override(zk.Widget.prototype, exWidget, {
+		bind_: function(dt, skipper, after){
+			exWidget.bind_.apply(this, arguments);
+			// Override the widget so that all components can act as drop areas
+			if (this.isFileDragDropArea){
+				jq(this).on("drop", this.onFileDrop.bind(this));
+				jq(this).on("dragover", this.onFileDragOver.bind(this));
+				jq(this).on("dragenter", this.onFileDragEnter.bind(this));
+				jq(this).on("dragleave", this.onDragLeave.bind(this));
+			}
+		},
+		onFileDrop: function(ev){
+			ev.preventDefault();
+			// clean border to indicate the drop area
+			jq(this).removeClass(this.cssDragEnter);
+			var dt = ev.originalEvent.dataTransfer;
+			if (dt.files && dt.files.length > 0){
+				// get upload component widget (component setup follow https://www.zkoss.org/wiki/ZK_Developer's_Reference/UI_Patterns/File_Upload_and_Download)
+				wgUploadBt = zk.Widget.$(this.uploadID);
+				// get reference to file input
+				var inp = wgUploadBt._uplder._inp;
+				// set FileList from drop to file input
+				inp.files = ev.originalEvent.dataTransfer.files;
+				// fire up change event to do upload
+				jq(inp).trigger("change");
+			}
+			
+			return false;
+		},
+		onFileDragOver: function(ev){
+			ev.preventDefault();
+			return false;
+		},
+		onFileDragEnter:function(ev) {
+			ev.preventDefault();
+			// highlight border to indicate the drop area
+			jq(this).addClass(this.cssDragEnter);
+			return false;
+		},
+		onDragLeave: function (ev){
+			ev.preventDefault();
+			if (ev.delegateTarget == ev.target){
+				// clean border to indicate the drop area
+				jq(this).removeClass(this.cssDragEnter);
+			}
+			return false;
+		},
+		cssDragEnter:"attachment-drag-entered",
+	});
```

Comment:
To avoid potential conflict with future zk update, probably safer to use custom prefix (for e.g `_id_`) for all custom property and function.